### PR TITLE
Add support for local build option

### PR DESCRIPTION
**Pull request by brienw**
Adds support for local builds, and refactors the way tasks are loaded.

The build task is now broken up into two additional sub-task files - one for local tasks, one for remote tasks. In order to get them all loading, I had to rework the way all bootleg tasks were loaded. From now on, all the `tasks/` directory is loaded recursively whereas before there was an assumption that the directory structure was flat and only contained files.

### CODE REVIEW

**Comment by brienw** [comment]
Fixes #193

File: README.md [hunk 1]
```diff
@@ -119,10 +119,11 @@ There are a few config settings that are directly used within Bootleg itself, th
 
 ```elixir
 config :app, :myapp
-config :refspec, "develop" # Set a git branch used for the build. Default is "master"
-config :version, "1.2.3"
 config :env, :staging # sets/overrides the bootleg environment
 config :ex_path, "/path/to/project" # Path to the project. Default is current directory.
+config :local_build # build releases locally without a `:build` role
```
Comment (rjanja) [inline README.md]: This should have a value supplied to it indicating it's a boolean option

File: README.md [hunk 2]
```diff
@@ -160,6 +161,17 @@ Some features or extensions may require additional roles, for example if your
 release needs to run Ecto migrations, you will need to assign the `:db`
 role to one host.
 
+#### Local build option
+
+You may not need a remote build server for all situations. In these cases, you can skip designating a `:build` role, and instead set the top level `:local_build` configuration to `true`. This will tell bootleg to skip all of the steps required to build the release on a remote server and instead built it locally.
```
Comment (rjanja) [inline README.md]: s/configuration/option

File: README.md [hunk 3]
```diff
@@ -160,6 +161,17 @@ Some features or extensions may require additional roles, for example if your
...
+
+```elixir
+# deploy.ex
```
Comment (rjanja) [inline README.md]: `deploy.exs`

File: lib/bootleg/tasks.ex [hunk 4]
```diff
@@ -90,6 +86,27 @@ defmodule Bootleg.Tasks do
...
+    end)
+  end
+
+  defp load_bootleg_task file do
+    Code.eval_string(
+      File.read!(file),
+      [],
+      %{__ENV__ | line: 1, file: file}
+    )
+  end
```
Comment (rjanja) [inline lib/bootleg/tasks.ex]: We also have `Config.load` for evaluating elixir script. Both these seem to do the same thing. We might combine them here, or defer it for later, but just calling it out.

File: test/bootleg/tasks/build_task_functional_test.exs [hunk 5]
```diff
@@ -39,17 +39,30 @@ defmodule Bootleg.Tasks.BuildTaskFunctionalTest do
...
     # credo:disable-for-next-line Credo.Check.Consistency.MultiAliasImportRequireUse
     use Bootleg.Config
-    # credo:disable-for-next-line Credo.Check.Consistency.MultiAliasImportRequireUse
-    alias Bootleg.Config
 
     File.cd!(location, fn ->
       capture_io(fn ->
         invoke :build
-        release_name = "#{Config.version()}.tar.gz"
+        release_name = "#{config :version}.tar.gz"
```
Comment (rjanja) [inline test/bootleg/tasks/build_task_functional_test.exs]: 👍

File: lib/bootleg/tasks.ex [hunk 6]
```diff
@@ -90,6 +86,27 @@ defmodule Bootleg.Tasks do
     end)
   end
 
+  defp load_bootleg_tasks path do
```
Comment (rjanja) [inline lib/bootleg/tasks.ex]: Is this a new style guide/code formatting syntax? We might consider following the same format we use elsewhere in the project, or updating it everywhere in a separate commit

File: lib/bootleg/tasks.ex (hunk 4 above)
Comment (brienw) [reply lib/bootleg/tasks.ex]: This is one of the first refactor targets. We actually load in a few different spots across bootleg, (two just inside this file!) so i want to see if there is a a way we can combine at least a few of them. I know we have to load differently depending on whether it's an internal vs external task, but maybe there's still hope to unify everything under one

File: lib/bootleg/tasks.ex (hunk 6 above)
Comment (brienw) [reply lib/bootleg/tasks.ex]: oops, oversight on my part, should have parens - especially now that with formatter it's The Elixir Way(TM)

Comment (brienw) [reply lib/bootleg/tasks.ex]: related: after this is merged we're gonna run `mix formatter` on the whole shebang

Comment (brienw) [reply lib/bootleg/tasks.ex]: also odd that credo didn't notice this inconsistency, hmmm

**Pull request merged**